pinot-core/src/main/java/org/apache/pinot/core/segment/creator/GeoSpatialIndexCreator.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.creator;
+
+import java.io.Closeable;
+
+
+/**
+ * An index creator for geospatial data
+ */
+public interface GeoSpatialIndexCreator extends Closeable {
+
+  /**
+   * Adds a doc associated with a geospatial point
+   * @param docId the document id
+   * @param lat the latitude
+   * @param lon the longitude
+   */
+  void add(int docId, double lat, double lon);
```

Comment:
We don't usually put docId explicitly. It is always appended as the next doc

Reply:
But how does that work? The index creator keeps the mapping of `H3Index -> BitMap of docIds`